### DISCUSSION TITLE
Handling Large Numbers of Dependent Jobs in BullMQ - Best Practices?

### DISCUSSION THREAD

**Discussion by jgalbsss** [taskforcesh/bullmq#2933] [Q&A]
I'm working with BullMQ in a Node.js application and need advice on handling job dependencies at scale. Here's my scenario:
I have a parent job that needs to spawn multiple child jobs (potentially up to 3000)
The parent job needs to wait for all child jobs to complete before proceeding
Need a solution that can handle this volume of dependent jobs reliably
What's the recommended pattern for handling large numbers of dependent jobs while:
Avoiding event listener limits
Maintaining good performance
Ensuring reliable job completion tracking
Has anyone solved this at scale? Looking for battle-tested approaches or best practices.

**Comment by manast** [comment]
@roggervalf will probably chime in with more insights, but basically the performance for child jobs is not affected by the number of jobs, or anything like that, so from a performance perspective there will be no bottlenecks that I am aware of. Regarding the "event listener limit" I am not sure what are you referring to here, could you expand a bit more on it?

**Reply by roggervalf** [reply]
not aware of bottlenecks in this case and 3000 children doesn't sound like a big number for this case in my opinion
